client/utils/sort-challengefiles.ts

```diff
@@ -0,0 +1,19 @@
+export function sortChallengeFiles<File extends { fileKey: string }>(
+  challengeFiles: File[]
+): File[] {
+  const xs = challengeFiles.slice();
+  xs.sort((a, b) => {
+    if (a.fileKey === 'indexjsx') return -1;
+    if (b.fileKey === 'indexjsx') return 1;
+    if (a.fileKey === 'indexhtml') return -1;
+    if (b.fileKey === 'indexhtml') return 1;
+    if (a.fileKey === 'stylescss') return -1;
+    if (b.fileKey === 'stylescss') return 1;
+    if (a.fileKey === 'scriptjs') return -1;
+    if (b.fileKey === 'scriptjs') return 1;
+    if (a.fileKey === 'indexts') return -1;
+    if (b.fileKey === 'indexts') return 1;
+    return 0;
+  });
+  return xs;
```

Comment:
```suggestion
  return challengeFiles.toSorted((a, b) => {
    if (a.fileKey === 'indexjsx') return -1;
    if (b.fileKey === 'indexjsx') return 1;
    if (a.fileKey === 'indexhtml') return -1;
    if (b.fileKey === 'indexhtml') return 1;
    if (a.fileKey === 'stylescss') return -1;
    if (b.fileKey === 'stylescss') return 1;
    if (a.fileKey === 'scriptjs') return -1;
    if (b.fileKey === 'scriptjs') return 1;
    if (a.fileKey === 'indexts') return -1;
    if (b.fileKey === 'indexts') return 1;
    return 0;
  });
```

This is a total nit, but does our environment support `toSorted`?